lib/assessment.js

```diff
@@ -32,7 +40,24 @@ var any = function any(array, callback, thisArg) {
 // made up of the array elements for which invoking
 // the callback returns a truthy value
 var select = function select(array, callback, thisArg) {
+  var newArray = [];
+  var innerCallback = function(currentValue, index, array) {
+
+    newArray.push(callback(currentValue, index, array));
+  };
+
+  array.forEach(innerCallback);
+
 
+
+  return newArray;
+};
```

Comment:
Close but not quite. We only want to push a given value into the new array if `callback(value)` is true. Can you see how your code might be modified to get that result?